### DISCUSSION TITLE
New Action for Tasks in Schedules: Update Startup Variables

### DISCUSSION THREAD

**Discussion by Maelep** [pterodactyl/panel#3699] [Feature Requests]
Add the `Update Startup Variables` option in the Action drop down menu when creating a new task that allow user to update a variables they have permission to.
This action displays a new drop-down menu with the variables that the user is allowed to modify. In addition, this action displays a text box to list the available values. A random value from the list will then be chosen. For example, for the variable `Map Seed`:
```txt
4
54
898
154654
546556465
```

**Comment by parkervcp** [comment]
I have already worked around something like this in my rust auto-wipe egg.

**Reply by Maelep** [reply]
I saw your workaround, it's a good one but unfortunately it doesn't allow to automate things. The current big problem concerns Rust and the management of wipes.

If we had at least the two suggestions I made then this problem would be solved.

Moreover, this feature would also allow to automate the change of server names, banners according to the current season or event, ... and not only for Rust but for all games and software.

**Reply by parkervcp** [reply]
in theory I could write something in the startup/install that would work on a specific day of the week.

**Reply by Maelep** [reply]
I fully rewrote the Docker image, the egg and the actual wrapper to make it more complete and functional for Rust (only) which allowed me to do something similar (wipe based on a given date) but again, this is a workaround. It works but it's not really practical, stable and user-friendly, the slightest error can wipe the server when you don't want it to xD

**Comment by trenutoo** [comment]
A drop-down menu of selectable values is coming in V2. A more fitting feature that you're looking for is the ~[scheduled file removal task](https://github.com/pterodactyl/panel/issues/2422)~ or [pre-startup scripts.](https://github.com/pterodactyl/panel/issues/1705) instead of something that's a game-specific feature

**Reply by Maelep** [reply]
My suggestion isn't specific to Rust but can be use by any other software/games. I don't have example for now.

Taking example for Rust, the problem isn't changing the seed of the game (I can easily do that with a script) but the problem is saving the value. I mean I change the seed using my script, I can't save the startup variable in the user panel, I'll have to store it with a script in the server files and so on.

By adding this possibility to directly change a value of a startup variable, I mean updating the value of the startup value "forever", like I changed this variable using the user panel > Startup.

**Reply by trenutoo** [reply]
At this moment, you can replace the seed in the startup itself by using a rand generation as an example. That's the value passed to the server to use for the seed.

I personally can't find many other uses for this scheduled task besides map randomization.

**Reply by Maelep** [reply]
The problem generating a random seed without saving it means the server will generate a new map every restart so it basically wipe every restart/crash/...

- You could name the server "Fresh wipe" at wipe and then, 3 days later, you could rename it.
- For software/game that use banner, you could automatically set a banner depending the season (like for halloween).
- Updating gamemode of some games.
- ...
